numba_dppy/compiler.py

```diff
@@ -444,7 +446,8 @@ def _pack_argument(self, ty, val, sycl_queue, device_arr, access_type):
             # device_array or if access_type of this device_array
             # is not of type read_only and read_write
             usm_buf, usm_ndarr, orig_ndarray = device_arr
-            np.copyto(orig_ndarray, usm_ndarr)
+            if isinstance(orig_ndarray, np.ndarray):
+                np.copyto(orig_ndarray, usm_ndarr)
```

Comment:
The `DPCTLQueue_Memcpy` in parfor is using the `C` function which can work with `void *`. The `DPCTLQueue_Memcpy` we have to use here has to be the `Python` API which expects the source and destination to be of type `_Memory`. The intention here is to avoid creating a `_Memory` object wrapping the `ndarray` data buffer, I am not sure if that is possible either.